pkg/eval/json_evaluator_test.go

```diff
@@ -0,0 +1,342 @@
+package eval
```

Comment:
I am a heavy user of `github.com/stretchr/testify/assert` so makes totally sense to me 👍

Reply:
I did not go into details in your tests but in **golang** there is heavy use of **table-driven style** to write tests.

You can check this example https://gobyexample.com/testing to see what I mean.
It makes tests way more readable since you have one function only for everything you want to test.

You can check this [file](https://github.com/thomaspoignant/go-feature-flag/blob/main/internal/flagv1/rollout_test.go) in one of my repo to see how it look like.

I will go more into what you are doing inside the tests later.

Reply:
@thomaspoignant I was wrong. I made the tests for the resolver methods tabular, and I like it much better. It also increased test coverage.